src/LibraryUtilities.ts

```diff
@@ -295,7 +359,78 @@ export function buildLibraryItemsFromLayoutSpecs(loadedTypes: any, layoutSpecs:
     for (let i = 0; i < layoutSpecs.elements.length; i++) {
         layoutElements.push(new LayoutElement(layoutSpecs.elements[i]));
     }
-    return convertToLibraryTree(typeListNodes, layoutElements);
+
+    let libraryTreeItems: ItemData[] = convertToLibraryTree(typeListNodes, layoutElements);
+    let unprocessedNodes: TypeListNode[] = [];
+
+    // Search for the nodes that are not displayed in library view.
+    _.each(typeListNodes, function (node) {
+        if (!node.processed) {
+            console.warn("Item filtered out from library view: " + node.contextData);
+            unprocessedNodes.push(node);
+        }
+    });
+
+    let miscNode = convertToMiscNodes(unprocessedNodes);
+    miscNode.text = "Miscellaneous";
+    
+    // Change the itemType of the outermost parents
+    _.each(miscNode.childItems, function(node) {
+        if (node.itemType == "group") node.itemType = "category";
+    })
+
+    libraryTreeItems.push(miscNode);
+    return libraryTreeItems;
+}
+
+/**
+ * Convert an array of typeListNodes to ItemData based on their fullyQualifiedNames.
```

Comment:
If this method converts type list nodes to item data, why is it called `convertToMiscNodes`? We need example data in description (and basically everywhere else) to make the code understandable.